src/pages/List.js

```diff
@@ -221,26 +223,30 @@ export default function List({ token }) {
       <h1>This Is Your Grocery List</h1>
       <h2>It uses the token: {token}</h2>
       <label htmlFor="thesearch">Search Grocery List Items </label>
-      <input
-        type="text"
-        placeholder="enter grocery item"
-        value={query}
-        id="thesearch"
-        onChange={(e) => setQuery(e.target.value)}
-      />
-      <button onClick={handleReset}>Reset Text Field</button>
+      <div className="flex">
+        <input
+          type="text"
+          placeholder="enter grocery item"
+          value={query}
+          id="thesearch"
+          onChange={(e) => setQuery(e.target.value)}
+        />
+        <IconButton onClick={handleReset} text="X" />
+      </div>
 
       {error && <strong>Error: {JSON.stringify(error)}</strong>}
       {loading && <span>Grocery List: Loading...</span>}
       {listItems && (
         <>
           <h2>Grocery List:</h2>
           {listItems.docs.length === 0 ? (
-            <section>
+            <section className="flex flex-col items-center w-full">
               <p>Your grocery list is currently empty.</p>
-              <button onClick={() => history.push('/add-item')}>
-                Click to add first item!
-              </button>
+              <Button
+                onClick={() => history.push('/add-item')}
+                text="+ Click to add your first item"
+                tailwind="mt-5"
```

Comment:
What does this line `tailwind="mt-5"` mean?

Reply:
I wanted to demonstrate that if you need to add a utility class you can pass it in as props, referred to as `tailwind`. If that name is confusing, we can change that! But I kept the styling to the button as minimal and self-contained as possible, and not making guesses at what it's specific styles might require based on its usage.

Do you think this is an unneccessary addition?

Reply:
The name is confusing because I am not sure what I am allowed to pass to it. See my other comment above. Allowing customization could be good and should be done in a way where it's flexible and optional (I see an undefined in your code). A better name could be `customStyles`, etc depending on what's allowed as overrides.

![image](https://user-images.githubusercontent.com/6759658/119362105-36130d80-bc61-11eb-8eb1-e3484ec7ae25.png)

Reply:
I see what you mean, @yenly. Maybe a better name would something like `additionalUtilityClasses` because it is only intended to add extra tailwind utility classes. Or if `additionalTailwindUtilityClasses` isn't too verbose, maybe that's self-explanatory enough?

As far as the undefined goes, that's strange. I need to dig into that. It works as intended on the `<Button>` that passes in `mt-5`. But that button you screen-shot does not have any additional utility classes being passed in, and I am guessing that's where the `undefined` is coming from. I have an idea how to fix it, but I'll confirm in just a moment!

Reply:
I have updated the `Button.js` and `IconButton.js` componets to remove the additional styling prop `tailwind` and cleaned up the pages where those buttons were called, removing their `tailwind` properties as well. 

After review and further reflection, I think it makes more sense to not burden the developer with passing in empty props `tailwind=""` and instead allow them to focus on apply appropriate styles to neighboring elements.